nexus/server.go

```diff
@@ -26,6 +26,21 @@ type handlerCtx struct {
 	links []Link
 }
 
+// WithHandlerContext returns a new context from a given context setting it up for being used for handler methods.
+// Meant to be used by frameworks, not directly by applications.
+func WithHandlerContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, handlerCtxKey, &handlerCtx{})
+}
+
+// HandlerLinks retrieves the attached links on the given handler context. The returned slice should not be mutated.
+// The context provided must be the context passed to the handler.
+func HandlerLinks(ctx context.Context) []Link {
+	hctx := ctx.Value(handlerCtxKey).(*handlerCtx)
+	hctx.mu.Lock()
+	defer hctx.mu.Unlock()
+	return hctx.links
```

Comment:
I think these should be copied on return (new slice/array + https://pkg.go.dev/builtin#copy is quite cheap especially for small uses) and there should be a set links too

Reply:
Why? Are you worried about users mutating? Is it typical for Go libraries to protect users this way?

Reply:
Yes and yes, e.g. https://github.com/grpc/grpc-go/blob/v1.70.0/metadata/metadata.go#L197. This is a very common approach for returning mutable slices that are rarely asked for.